tests/org.eclipse.n4js.ui.tests/src/org/eclipse/n4js/tests/externalPackages/GH_986_IntroduceExternalCircularDependencyAndRunTest.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) 2018 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.tests.externalPackages;
+
+import java.util.Set;
+
+import javax.inject.Provider;
+
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.MultiStatus;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.n4js.n4mf.ProjectType;
+import org.eclipse.n4js.projectModel.IN4JSCore;
+import org.eclipse.n4js.projectModel.IN4JSProject;
+import org.eclipse.n4js.runner.RuntimeEnvironmentsHelper;
+import org.eclipse.n4js.runner.extension.RuntimeEnvironment;
+import org.eclipse.n4js.tests.builder.AbstractBuilderParticipantTest;
+import org.eclipse.n4js.tests.util.ShippedCodeInitializeTestHelper;
+import org.eclipse.n4js.ui.external.ExternalLibrariesActionsHelper;
+import org.eclipse.n4js.ui.wizard.dependencies.InstallOptions;
+import org.eclipse.n4js.ui.wizard.dependencies.RunnableInstallDependencies;
+import org.eclipse.n4js.utils.StatusHelper;
+import org.eclipse.xtext.ui.testing.util.IResourcesSetupUtil;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.google.common.base.Optional;
+import com.google.inject.Inject;
+
+/**
+ * Installs {@code mocha} and {@code eslint} as project dependency, which will introduce a dependency cycle in the
+ * dependency tree of the test project ({@code eslint} and {@code mocha} depend on each other via
+ * {@code devDependencies}).
```

Comment:
good to have a test for this!
But could we add a second test with a dependency cycle in the main dependencies (not devDependencies).
Rationale: we might change how we handle `devDependencies` and then this test might become ineffective.